pennylane/ops/qubit/state_preparation.py

```diff
@@ -156,6 +158,32 @@ def compute_decomposition(state, wires):
         """
         return [MottonenStatePreparation(state, wires)]
 
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_matrix(self):
+        return True
+
+    def matrix(self, wire_order=None):
+        """Returns a ket vector representing the state being created.
+
+        QSV([1/2, 1/2, 1/2, 1/2], wires=[0,2]) = 1/4 * (|0 0> + |0 1> + |1 0> + |1 1>)
+                                               = 1/4 * (|000> + |001> + |100> + |101>)
+        """
+        if wire_order is None:
+            return np.array(self.parameters[0], dtype="complex128")
```

Comment:
@timmysilv how come you are casting to NumPy here? I am thinking of a use case where someone has

```python
QubitStateVector(torch.tensor(...), wires=...)
```

Reply:
the goal was more to cast to complex128. My original plan was for this to be consumed by a particular state-prep procedure with a `like=some_interface` argument, but it seems like that won't be all encompassing so I'll support the preservation of interfaces here (as I'm doing in the linked BasisState PR now as well)